src/stx/common/trex_rx_timesync.cpp

```diff
@@ -8,89 +8,92 @@
 
 void RXTimesync::handle_pkt(const rte_mbuf_t *m, int port) {
     if (m_timesync_engine->getTimesyncMethod() == TimesyncMethod::PTP) {
-        uint16_t rx_tstamp_idx = 0;
 
-        if (hardware_timestamping_enabled) {
-            rx_tstamp_idx = m->timesync;
-        }
+        timespec t { 0, 0 };
 
         if (m->pkt_len < ETH_HDR_LEN)
             return;
 
-        uint64_t m_timestamp = 0;
-        if (CGlobalInfo::m_options.is_timesync_rx_callback_enabled()) {
-            m_timestamp = m->timestamp;
+        if (hardware_timestamping_enabled && (m->ol_flags & PKT_RX_IEEE1588_TMST)) {
+            if (rte_eth_timesync_read_rx_timestamp(port, &t, m->timesync) != 0) {
+                printf("Cannot read hardware timestamp from hardware");
+            }
+        } else if (CGlobalInfo::m_options.is_timesync_rx_callback_enabled()) {
+            t = timestampToTimespec(m->timestamp);
+        } else {
+            if (clock_gettime(CLOCK_REALTIME, &t) != 0) {
+                printf("Something is wrong!!! Cannot read time from kernel!");
+            }
         }
+
         uint8_t *pkt = rte_pktmbuf_mtod(m, uint8_t *);
         EthernetHeader *ether_hdr = (EthernetHeader *)pkt;
         uint32_t offset = ether_hdr->getSize();
         switch (ether_hdr->getNextProtocol()) {
         case EthernetHeader::Protocol::IP:
-            parse_ip_pkt(pkt + offset, m->pkt_len - offset, rx_tstamp_idx, port, m_timestamp);
+            parse_ip_pkt(pkt + offset, m->pkt_len - offset, &t, port);
             break;
         case EthernetHeader::Protocol::PTP:
             // TODO what about vxlan support (a.k.a. CFlowStatParser.m_flags |= FSTAT_PARSER_VXLAN_SKIP)
-            parse_ptp_pkt(pkt + offset, m->pkt_len - offset, rx_tstamp_idx, port, m_timestamp);
+            parse_ptp_pkt(pkt + offset, m->pkt_len - offset, &t, port);
             break;
         }
     }
 }
 
-TimesyncPacketParser_err_t RXTimesync::parse_ip_pkt(uint8_t *pkt, uint16_t len, uint16_t rx_tstamp_idx, int port, uint64_t m_timestamp) {
+TimesyncPacketParser_err_t RXTimesync::parse_ip_pkt(uint8_t *pkt, uint16_t len, timespec* t, int port) {
```

Comment:
nit: wouldn't putting `port` before `t` be a little bit more consistent? (at least for me it would :stuck_out_tongue_winking_eye: )